examples/dfk2wan.jl

```diff
@@ -0,0 +1,149 @@
+# description of the system, as well as three data files `.mmn`,
+# `.amn` and `.eig`. These files are respectively generated by
+# the functions `dtk2wan_win_file` and
+# `dftk2wan_wannierization_files`. The whole procedure
+# takes five steps :
+#
+#     - SCF calculation
+#     - call to dftk2wan_win_file
+#     - Wannier90's preprocessing step
+#     - call to dftk2wan_wannierization_files
+#     - Wannier90's computation of MLWFs
+#
+# This example shows how to obtain the MLWFs corresponding
+# to the first eight bands of silicon. Since the bands 5 to 8 are entangled,
+# 12 bands are first computed to obtain 8 MLWFs by a disantanglement procedure.
+# We first perfom a SCF calculation, which is for the most part identical to the
+# regular case.
+#
+# !!! warning "Compatibility asks for a specific setup"
+#     The `.win` file will
+#     correspond to the studied system only if the number of ``k``
+#     points and bands in input is the same as in output.
+#     Since DFTK reduces by default
+#     the number of ``k`` points by symmetry, and adds automaticaly
+#     three non-converged bands, one must specify :
+#     - `use_symmetry = false` in the creation of the plane wave basis
+#     - `n_ep_extra = 0` in the self_consistent_field declaration.
+#
+
+using DFTK
+
+a = 10.26
+lattice = a / 2*[[-1.  0. -1.];
+                 [ 0   1.  1.];
+                 [ 1   1.  0.]]
+
+Si = ElementPsp(:Si, psp=load_psp("hgh/pbe/Si-q4"))
+atoms = [ Si => [zeros(3), 0.25*[-1,3,-1]] ]
+
+model = model_PBE(lattice,atoms)
+
+kgrid = [4,4,4]
+Ecut = 20
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid, use_symmetry=false)
+
+scfres = self_consistent_field(basis, tol=1e-12, n_bands = 12, n_ep_extra = 0);
+
+# We may now create the `.win` file. In addition of informations on the system
+# (via `basis` and `scfres`) the functon asks for a prefix (here "Si") and
+# the number of wanted MLWFs; 8 in our case. The keyword `bands_plot = true`
+# adds the ``k`` point path to the `.win` file, needed to plot the bands structure.
+# We also add as keyword arguments the parameters of the disentanglement
+# procedure. Any other argument can be added as such.
+
+prefix = "Si"
+dftk2wan_win_file(prefix, basis, scfres, 8;
+                  bands_plot=true, num_print_cycles=50, num_iter=500,
+                  dis_win_max       = "17.185257d0",
+                  dis_froz_max      =  "6.8318033d0",
+                  dis_num_iter      =  120,
+                  dis_mix_ratio     = "1d0")
+
+# We then perfom a preproprocessing task via Wannier90 to generate
+# the `Si.nnkp` file. This is simply done by :
+
+#md #```@raw bash
+#md #$ PathToWannier90/wannier90.x -pp Si
+
+# !!! note "Launching Wannier90 from Julia file or shell"
+#     All Wannier90 calls can be done through julia, with
+#     the help of the package wannier90_jll. The previous
+#     command is written as follows
+
+using wannier90_jll
+wannier90() do exe
+    run(`$exe -pp Si`)
+end
```

Comment:
Also if this is the workflow, why is there no function in DFTK that does the whole lot? Is there a reason why the user has to "see" the individual steps?

Reply:
>  Why can't that just be wannier90("-pp Si")?

Because the `wannier90_jll` actually messes with the environment variables to ensure that wannier90 and dependent libraries are found. So if we do not want to hide this call inside DFTK than this is the way it has to be done.

Reply:
The wannier90 workflow is sufficiently messy and error-prone that it's good to leave the black box open here. Also we might want to replace parts of wannier90 (like all of it) incrementally, so let's not commit ourselves to something final just yet

Reply:
I see. Ok. Still perhaps we might want to have a nice support function that hides at least the wannier90_jll ... don't know just a suggestion.

Reply:
I don't understand. What's keeping me from defining `wannier90(opts::String) = wannier90(exe -> run(`$exe $opts`)`?

Reply:
meh github doesn't like nested backticks, but you understand what I mean

Reply:
In fact I chose to work that way so the user can control the win file and add what's needed. I think we choose:
 1) to hide all the nnkp creation and creation of amn, mmn, eig, but every options has to be called as kwargs.
 2) keep the process split in several steps to be able to control them individually. 

Since I didn't use much Wannier90 I don't quite realize how badly the calculation can turn if you cannot inspect every step. Nevertheless, with wannier90__jll the option 1 can be done very easily I think.

Reply:
Also if we choose the option 1) I can remove all the formating process of the win file to reduce the length of the parser (even if I have to admit I love well parsed files).

Reply:
Oh in DFTK nothing, but that is not the right way to do it ;). It should be:
```
wannier90(opts::Cmd) = wannier90_jll.wannier90(exe -> run(`$exe $opts`)
```
To do with shell quoting :smile:

Reply:
Yeah right, try a file name with spaces and see how wannier90 likes it :-p Anyway, I don't know but this command thing just feels like a very roundabout API for running a command.

Reply:
Well it's low-level. That's the point of the jll packages. If you don't like it ... write a wrapper! That's the point.

Reply:
I don't understand what's going here, but that stuff passes me over the head. If it's needed, might as well hide it : by defining a DFTK.run_wannier90(options) that calls `split` on the input and passes it to the wannier90_jll

Reply:
To clarify above: I'm not bitching about the wannier90_jll, which I think is awesome, I'm bitching about why we have to pass a closure to run a command line program

Reply:
I also think we misunderstand. Just to be certain: You mean in the example or in general to run the jll?

I'm 0% against having a `run_wannier90` function that takes some args, processes them to the extent needed and then runs wannier.